src/libs/actions/IOU.ts

```diff
@@ -551,6 +564,18 @@ function buildOnyxDataForMoneyRequest(
         },
     );
 
+    if (!isOneOnOneSplit) {
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.SET,
+            key: ONYXKEYS.NVP_QUICK_ACTION_GLOBAL_CREATE,
+            value: {
+                action: newQuickAction,
+                reportID: chatReport?.reportID,
+                isFirstQuickAction: isEmptyObject(quickAction),
+            },
+        });
+    }
```

Comment:
Coming from https://github.com/Expensify/App/issues/50311, here we missed the `failureData` that will revert the quick action to the previous one when the current action fails like we do here https://github.com/Expensify/App/blob/6d8e739f7b1c286ae65565a6689b27b5983fe48a/src/libs/actions/IOU.ts#L1540-L1544
and it fixed here https://github.com/Expensify/App/pull/50617